plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -201,8 +201,12 @@ public void initModel(final Connection connection, final IProject project, boole
 				protected IStatus doRun(IProgressMonitor monitor) {
 					try {
 						List<IProject> projects = connection.getResources(ResourceKind.PROJECT);
+						//Project might be set by a master model that set connection to this slave model
+						//and thus caused loading projects job. While this job was in wait,
+						//master model might set project. It must be respected.
```

Comment:
Yes I get that, your solution is simple enogh and effective. My concern is that we have a design flaw that we need to address sooner or later. 
The current solution doesnt look deterministic enough to me. I also dislike the simple fact that we have a method #initModel that takes a project argument which is then eventually ignored. This is imho the kind of code that causes problems in maintenance because it's very unexpected. We should correct this as long as we still understand the context and the problem. It'll get much harder to fix later.